packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -136,15 +267,20 @@ contract DowntimeSlasher is SlasherUtil {
     address[] memory groupElectionGreaters,
     uint256[] memory groupElectionIndices
   ) public {
-    address validator = getAccounts().signerToAccount(
-      validatorSignerAddressFromSet(startSignerIndex, startBlock)
+    require(startBlocks.length > 0, "requires at least one interval");
+    require(signerIndices.length > 0, "requires at least one signerIndex");
+    address validator = getValidatorAccountFromSignerIndex(signerIndices[0], startBlocks[0]);
+    uint256 endBlock = startBlocks[0].add(slashableDowntime).sub(1);
```

Comment:
Consider the following minor adjustment:
```
uint256 startBlock = startBlocks[0];
uint256 endBlock = endBlocks[-1];
require(endBlock.sub(startBlock) >= slashableDowntime, "the provided intervals must span slashableDowntime blocks");
address validator = getValidatorAccountFromSignerIndex(signerIndices[0], startBlock);
require(startBlock >= lastSlashedBlock[validator], "cannot slash validator for downtime for which they may already have been slashed");
require(wasDownForIntervals(startBlocks, endBlocks, signerIndices), "not down");
lastSlashedBlock[validator] = endBlock;
performSlashing();
```

Reply:
I was doing
```
uint256 endBlock = startBlocks[0].add(slashableDowntime).sub(1);
```
instead of
```
uint256 endBlock = endBlocks[-1];
```
To avoid the user to "cover" on purpose part of another slashable interval, but now that we only check that the slash should be "newer" it doesn't have much sense... The user could cover all with the newest interval

Reply:
I'm adding this minor change, `startBlock > lastSlashedBlock[validator]`, instead of `>=`

Reply:
and
```
require(endBlock.sub(startBlock).add(1) >= slashableDowntime, "the provided intervals must span slashableDowntime blocks");
```